examples/all-clusters-app/all-clusters-common/src/operational-state-delegates.cpp

```diff
@@ -0,0 +1,184 @@
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/EventLogging.h>
+#include <app/clusters/operational-state-server/operational-state-server.h>
+#include <app/reporting/reporting.h>
+#include <app/util/config.h>
+#include <operational-state-delegate-impl.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace OperationalState {
+
+/**
+ * class to present Enquriy Table of Delegates
+ */
+struct DelegatesEnquiryTable
+{
+    /**
+     * Endpoint Id
+     */
+    EndpointId mEndpointId;
+    /**
+     * Cluster Id
+     */
+    ClusterId mClusterId;
+    /**
+     * point of Array(Items)
+     */
+    Delegate * pItems;
+    /**
+     * ArraySize of Array(Items)
+     */
+    size_t numOfItems;
+};
+
+/*
+ * An example to present device's endpointId
+ */
+constexpr EndpointId kDemoEndpointId = 1;
+
+/**
+ * Enquriy Table of Operational State List
+ * Note: User Define
+ */
+static const GenericOperationalState opStateList[] = {
+    GenericOperationalState(to_underlying(OperationalStateEnum::kStopped)),
+    GenericOperationalState(to_underlying(OperationalStateEnum::kRunning)),
+    GenericOperationalState(to_underlying(OperationalStateEnum::kPaused)),
+    GenericOperationalState(to_underlying(OperationalStateEnum::kError)),
+};
+
+/**
+ * Enquriy Table of Phase List
+ * Note: User Define
+ */
+static const GenericOperationalPhase opPhaseList[] = {
+    /**
+     * Phase List is null
+     */
+    GenericOperationalPhase(DataModel::Nullable<CharSpan>()),
+};
+
+/**
+ * Enquriy Table of Operational State Delegate
+ * Note: User Define
+ */
+static OperationalStateDelegate opStateDelegate(Clusters::OperationalState::kDemoEndpointId, Clusters::OperationalState::Id,
+                                                GenericOperationalState(to_underlying(OperationalStateEnum::kStopped)),
+                                                GenericOperationalError(to_underlying(ErrorStateEnum::kNoError)),
+                                                Span<const GenericOperationalState>(opStateList),
+                                                Span<const GenericOperationalPhase>(opPhaseList));
+
+/**
+ * Enquriy Table of Operational State Cluster and alias Cluter Delegate corresponding to endpointId and clusterId
+ * Note: User Define
+ */
+constexpr DelegatesEnquiryTable kDelegatesEnquiryTable[] = {
+    // EndpointId, ClusterId, Delegate
+    { Clusters::OperationalState::kDemoEndpointId, Clusters::OperationalState::Id, &opStateDelegate },
```

Comment:
This is not setting `numOfItems`... thus reinforcing the "one delegate per endpoint/cluster" bit.